dbms/src/Interpreters/SettingsCommon.h

```diff
@@ -706,4 +706,57 @@ struct SettingString
     }
 };
 
+struct SettingChar
+{
+private:
+    void checkStringIsACharacter(const String & x) const {
+        if (x.size() != 1)
+            throw Exception(std::string("A setting's value string has to be an exactly one character long"));
```

Comment:
Missing ErrorCodes.

Reply:
Style. Braces `{}` should be in a separate new line.